src/api.c

```diff
@@ -2436,29 +2440,37 @@ static int _cgroup_create_cgroup(const struct cgroup * const cgroup,
 	fts_path[1] = NULL;
 	path = fts_path[0];
 
-	if (!cg_build_path(cgroup->name, path, controller->name))
-		goto err;
+	if (controller) {
+		if (!cg_build_path(cgroup->name, path, controller->name))
+			return 0;
```

Comment:
return here will cause a memory leak of `fts_path`, it should be replaced with `goto err;` and return in the else block below to needs to be `goto err`.

Reply:
D'oh.  Good catch.